controllers/snapshot/snapshot_adapter.go

```diff
@@ -523,7 +524,9 @@ func (a *Adapter) createSnapshotEnvironmentBindingForSnapshot(application *appli
 
 	for _, keyAndValue := range optionalLabelKeysAndValues {
 		if v, ok := keyAndValue[gitops.SnapshotTestScenarioLabel]; ok {
-			h.AddLabel(&snapshotEnvironmentBinding.ObjectMeta, gitops.SnapshotTestScenarioLabel, v)
+			temp_labels := map[string]string{}
```

Comment:
nit: `temp_labels` would probably be easier to understand as something like `newLabels`.

Reply:
I will update it.